tests/docker/Dockerfile

```diff
@@ -14,27 +14,33 @@
 # limitations under the License.
 FROM public.ecr.aws/docker/library/ubuntu:jammy-20240911.1 AS os_base
 FROM os_base AS base
+# set to any random value to rebuild all layers from the base on down
+ARG BASE_CACHE_BREAK=0
 # 1 to use the download-utils caching, 0 to disable
 ARG DL_USE_CACHE=1
 ENV TZ="UTC" \
     DEBIAN_FRONTEND=noninteractive
 
-RUN mkdir -p /opt/redpanda-tests
+# removing docker-clean to ensure the downloaded packages remain: we use a cache mount
+# for apt, so apt downloads and repo lists do not appear in the final image
+RUN rm /etc/apt/apt.conf.d/docker-clean && mkdir -p /opt/redpanda-tests
 COPY --chown=0:0 tests/protobuf /opt/redpanda-tests/protobuf
 
 COPY --chmod=0755 tests/docker/ducktape-deps/download-utils /
 COPY --chown=0:0 --chmod=0755 tests/docker/ducktape-deps/java-dev-tools /
 COPY --chown=0:0 --chmod=0755 tests/docker/ducktape-deps/protobuf /
-RUN --mount=type=cache,id=dl-cache,target=/dl-cache /java-dev-tools && \
-    rm /java-dev-tools && \
-    rm -rf /var/lib/apt/lists/*
```

Comment:
These `rm -rf apt/lists` can be removed, because (a) we want them now and (b) they are stored in the cache so they don't bloat the layer size which is generally the reason they are removed in the first place.